### PR TITLE
remove MoneyRequestCategoryPage and refactor route

### PR DESCRIPTION
**Pull request by dukenv0307**
### Details

1. Remove all use of `MoneyRequestCategoryPage`
2. Use `IOURequestStepCategory` for both create and edit case
3. Add `action` param for the route of `IOURequestStepCategory`
4. `EditRequestCategoryPage` will be removed after removing all use of `EditRequestPage`
### Fixed Issues

$ https://github.com/Expensify/App/issues/34606
PROPOSAL: https://github.com/Expensify/App/issues/34606#issuecomment-1896031635

### Tests

- create and edit request money
1. Verify that `MoneyRequestCategoryPage` is removed with the route, screen, and all use of this page in the app
2. Create a request money and select a category
3. Go to transaction thread report
4. Verify that selected category is displayed as well
5. Click on category to edit
6. Verify that the route has this format `:action/:iouType/category/:transactionID/:reportID` and action param in URL is `edit`
7. Select onther category and verify that category is edited successfully
- Create and edit scan split bill
1. Go to workspace chat
2. Create a scan split bill request and select a category
3. Click on split bill detail
4. Verify that selected category is displayed as well
5. Click on category to edit
6. Verify that the route has this format `:action/:iouType/category/:transactionID/:reportID` and action param in URL is `edit`
7. Select another category and verify that category is edited successfully
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as above
### QA Steps

- create and edit request money
1. Verify that `MoneyRequestCategoryPage` is removed with the route, screen, and all use of this page in the app
2. Create a request money and select a category
3. Go to transaction thread report
4. Verify that selected category is displayed as well
5. Click on category to edit
6. Verify that the route has this format `:action/:iouType/category/:transactionID/:reportID` and action param in URL is `edit`
7. Select another category and verify that category is edited successfully
- Create and edit scan split bill
1. Go to workspace chat
2. Create a scan split bill request and select a category
3. Click on split bill detail
4. Verify that selected category is displayed as well
5. Click on category to edit
6. Verify that the route has this format `:action/:iouType/category/:transactionID/:reportID?backTo=` and action param in URL is `edit`
7. Select another category and verify that category is edited successfully
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

- create and edit request money

https://github.com/Expensify/App/assets/129500732/59f6d176-cdd2-436b-9d08-538f5c609dfa

- Create and edit scan split bill

https://github.com/Expensify/App/assets/129500732/591c2273-427f-4aad-ac01-afcc7bcda7b5

</details>

<details>
<summary>Android: mWeb Chrome</summary>

- create and edit request money

https://github.com/Expensify/App/assets/129500732/dd614487-31ee-4862-90b7-0ad229f6c9af

- Create and edit scan split bill

https://github.com/Expensify/App/assets/129500732/e2752b52-8096-4359-be10-7544ea163e56

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

- create and edit request money

https://github.com/Expensify/App/assets/129500732/465db074-d6e3-42e1-bd0b-b1f1ac1f7c03

- Create and edit scan split bill

https://github.com/Expensify/App/assets/129500732/39833bb4-e1ea-48d8-b5b4-197ec3318e4b

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

- create and edit request money

https://github.com/Expensify/App/assets/129500732/9f35d3a7-21d0-429b-af5a-8d6cbe157da5

- Create and edit scan split bill

https://github.com/Expensify/App/assets/129500732/04e2bdb6-4f64-46ab-a909-f1c774f44b42

</details>

<details>
<summary>MacOS: Desktop</summary>

- create and edit request money

https://github.com/Expensify/App/assets/129500732/a08b0acd-e95c-4717-96b4-b10ce315421c

- Create and edit scan split bill

https://github.com/Expensify/App/assets/129500732/4987d352-2ecc-4083-a8bd-aa897fc3a971

</details>

### CODE REVIEW

**Comment by DylanDylann** [comment]
@dukenv0307 Please resolve conflict

File: src/libs/Navigation/linkingConfig/config.ts [hunk 1]
```diff
@@ -396,21 +396,6 @@ const config: LinkingOptions<RootStackParamList>['config'] = {
...
-                            [SCREENS.MONEY_REQUEST.CURRENCY]: ROUTES.MONEY_REQUEST_CURRENCY.route,
-                            [SCREENS.MONEY_REQUEST.DESCRIPTION]: ROUTES.MONEY_REQUEST_DESCRIPTION.route,
-                            [SCREENS.MONEY_REQUEST.TAG]: ROUTES.MONEY_REQUEST_TAG.route,
-                            [SCREENS.MONEY_REQUEST.MERCHANT]: ROUTES.MONEY_REQUEST_MERCHANT.route,
-                            [SCREENS.MONEY_REQUEST.RECEIPT]: ROUTES.MONEY_REQUEST_RECEIPT.route,
-                            [SCREENS.MONEY_REQUEST.DISTANCE]: ROUTES.MONEY_REQUEST_DISTANCE.route,
-                            [SCREENS.IOU_SEND.ENABLE_PAYMENTS]: ROUTES.IOU_SEND_ENABLE_PAYMENTS,
-                            [SCREENS.IOU_SEND.ADD_BANK_ACCOUNT]: ROUTES.IOU_SEND_ADD_BANK_ACCOUNT,
-                            [SCREENS.IOU_SEND.ADD_DEBIT_CARD]: ROUTES.IOU_SEND_ADD_DEBIT_CARD,
                         },
```
Comment (DylanDylann) [inline src/libs/Navigation/linkingConfig/config.ts:399]: @dukenv0307 Why change here?

Comment (DylanDylann) [reply src/libs/Navigation/linkingConfig/config.ts:399]: Ah, a bit laggy from Github 😄

File: src/pages/iou/request/step/IOURequestStepCategory.js [hunk 2]
```diff
@@ -50,10 +53,26 @@ function IOURequestStepCategory({
...
+            Navigation.dismissModal();
+            return;
+        }
+        // The case create request monney or split bill or send money
+        if (isSelectedCategory) {
             IOU.resetMoneyRequestCategory_temporaryForRefactor(transactionID);
         } else {
-            IOU.setMoneyRequestCategory_temporaryForRefactor(transactionID, category.searchText);
+            IOU.setMoneyRequestCategory_temporaryForRefactor(transactionID, updatedCategory);
         }
```
Comment (DylanDylann) [inline src/pages/iou/request/step/IOURequestStepCategory.js:76]: @dukenv0307  Please check this comment https://github.com/Expensify/App/issues/34608#issuecomment-1919249308.

**Comment by DylanDylann** [comment]
Also resolve conflict, please

**Comment by DylanDylann** [comment]
@dukenv0307 Please fix lint error and resolve conflict

File: src/libs/Navigation/types.ts [hunk 3]
```diff
@@ -221,10 +221,6 @@ type MoneyRequestNavigatorParamList = {
         field: string;
         threadReportID: string;
     };
-    [SCREENS.MONEY_REQUEST.CATEGORY]: {
-        iouType: string;
-        reportID: string;
-    };
     [SCREENS.MONEY_REQUEST.STEP_TAX_AMOUNT]: {
```
Comment (DylanDylann) [inline src/libs/Navigation/types.ts:224]: @dukenv0307 We should update correct type instead of removing it

File: src/libs/actions/IOU.ts [hunk 4]
```diff
@@ -3619,7 +3605,6 @@ function navigateToNextPage(iou: OnyxEntry<OnyxTypes.IOU>, iouType: string, repo
                 ? [{reportID: chatReport?.reportID, isPolicyExpenseChat: true, selected: true}]
                 : (chatReport?.participantAccountIDs ?? []).filter((accountID) => currentUserAccountID !== accountID).map((accountID) => ({accountID, selected: true}));
             setMoneyRequestParticipants(participants);
-            resetMoneyRequestCategory();
```
Comment (DylanDylann) [inline src/libs/actions/IOU.ts:3622]: Why do you remove this line?

File: src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsPage.js [hunk 5]
```diff
@@ -87,7 +87,6 @@ function MoneyRequestParticipantsPage({iou, selectedTab, route, transaction}) {
 
     const navigateToConfirmationStep = (moneyRequestType) => {
         IOU.setMoneyRequestId(moneyRequestType);
-        IOU.resetMoneyRequestCategory();
```
Comment (DylanDylann) [inline src/pages/iou/steps/MoneyRequstParticipantsPage/MoneyRequestParticipantsPage.js:90]: why do you remove this line?

**Comment by DylanDylann** [comment]
@dukenv0307 Please also use `IOURequestStepCategory` for editing requests and editing split bill

**Comment by DylanDylann** [comment]
Also should remove `EditRequestCategoryPage`

File: src/pages/iou/request/step/IOURequestStepCategory.js [hunk 6]
```diff
@@ -50,11 +53,22 @@ function IOURequestStepCategory({
...
+            navigateBack();
+            return;
+        }
+        // The casse edit request
+        if (isEditing) {
+            IOU.updateMoneyRequestCategory(transaction.transactionID, report.reportID, updatedCategory);
+            Navigation.dismissModal();
+            return;
         }
+        IOU.setMoneyRequestCategory(transactionID, updatedCategory);
```
Comment (DylanDylann) [inline src/pages/iou/request/step/IOURequestStepCategory.js:71]: I think we also should call setMoneyRequestCategory in the editing request flow to handle offline case. You can check [this PR](https://github.com/Expensify/App/pull/35137) for similar case

**Comment by DylanDylann** [comment]
@dukenv0307 I think we should remove `EditRequestCategoryPage` and all places that use this component to avoid confusion

**Comment by DylanDylann** [comment]
@dukenv0307 Any update here?

File: src/components/MoneyRequestConfirmationList.js [hunk 7]
```diff
@@ -766,11 +766,15 @@ function MoneyRequestConfirmationList(props) {
...
                             onPress={() => {
-                                if (props.isEditingSplitBill) {
-                                    Navigation.navigate(ROUTES.EDIT_SPLIT_BILL.getRoute(props.reportID, props.reportActionID, CONST.EDIT_REQUEST_FIELD.CATEGORY));
-                                    return;
-                                }
-                                Navigation.navigate(ROUTES.MONEY_REQUEST_CATEGORY.getRoute(props.iouType, props.reportID));
+                                Navigation.navigate(
+                                    ROUTES.MONEY_REQUEST_STEP_CATEGORY.getRoute(
+                                        CONST.IOU.ACTION.EDIT,
+                                        CONST.IOU.TYPE.SPLIT,
```
Comment (DylanDylann) [inline src/components/MoneyRequestConfirmationList.js:772]: NAB, should use iouType

File: src/pages/EditRequestPage.js [hunk 8]
```diff
@@ -170,16 +155,6 @@ function EditRequestPage({report, route, policy, policyCategories, policyTags, p
         );
     }
 
-    if (fieldToEdit === CONST.EDIT_REQUEST_FIELD.CATEGORY && shouldShowCategories) {
```
Comment (DylanDylann) [inline src/pages/EditRequestPage.js:173]: @dukenv0307 Make sure `shouldShowCategories` condition is migrated to IOURequestCategoryPage

File: src/pages/iou/request/step/IOURequestStepCategory.js [hunk 9]
```diff
@@ -21,24 +26,32 @@ const propTypes = {
...
-        params: {transactionID, backTo},
+        params: {transactionID, backTo, action, iouType},
     },
     transaction,
+    splitDraftTransaction,
 }) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
+    const isEditing = action === CONST.IOU.ACTION.EDIT;
+    const isSplitBill = iouType === CONST.IOU.TYPE.SPLIT;
```
Comment (DylanDylann) [inline src/pages/iou/request/step/IOURequestStepCategory.js:53]: Let's remove this variable and create isEditingSplitBill variable

File: src/pages/iou/request/step/IOURequestStepCategory.js [hunk 10]
```diff
@@ -49,11 +62,22 @@ function IOURequestStepCategory({
...
      */
     const updateCategory = (category) => {
-        if (category.searchText === transaction.category) {
-            IOU.resetMoneyRequestCategory_temporaryForRefactor(transactionID);
-        } else {
-            IOU.setMoneyRequestCategory_temporaryForRefactor(transactionID, category.searchText);
+        const isSelectedCategory = category.searchText === transactionCategory;
+        const updatedCategory = isSelectedCategory ? '' : category.searchText;
+
+        // The case edit split bill
```
Comment (DylanDylann) [inline src/pages/iou/request/step/IOURequestStepCategory.js:68]: Update to "In the split flow, when editing we use SPLIT_TRANSACTION_DRAFT to save draft value"

File: src/pages/iou/request/step/IOURequestStepCategory.js [hunk 11]
```diff
@@ -49,11 +62,22 @@ function IOURequestStepCategory({
...
+        if (isSplitBill && isEditing) {
```
Comment (DylanDylann) [inline src/pages/iou/request/step/IOURequestStepCategory.js:69]: Using isEditingSplitBill

File: src/pages/iou/request/step/IOURequestStepCategory.js [hunk 12]
```diff
@@ -49,11 +62,22 @@ function IOURequestStepCategory({
...
+            IOU.setDraftSplitTransaction(transaction.transactionID, {category: category.searchText});
```
Comment (DylanDylann) [inline src/pages/iou/request/step/IOURequestStepCategory.js:70]: Do we allow disable category in this case? Why don't you use updatedCategory here?

File: src/pages/iou/request/step/IOURequestStepCategory.js [hunk 13]
```diff
@@ -49,11 +62,22 @@ function IOURequestStepCategory({
...
+            navigateBack();
+            return;
+        }
+        // The casse edit request
```
Comment (DylanDylann) [inline src/pages/iou/request/step/IOURequestStepCategory.js:74]: remove this comment

File: src/pages/iou/request/step/IOURequestStepCategory.js [hunk 14]
```diff
@@ -49,11 +62,23 @@ function IOURequestStepCategory({
...
+
+        // In the split flow, when editing we use SPLIT_TRANSACTION_DRAFT to save draft value
+        if (isEditingSplitBill) {
+            IOU.setDraftSplitTransaction(transaction.transactionID, {category: category.searchText});
+            navigateBack();
+            return;
         }
+
+        if (isEditing) {
+            IOU.updateMoneyRequestCategory(transaction.transactionID, report.reportID, updatedCategory);
```
Comment (DylanDylann) [inline src/pages/iou/request/step/IOURequestStepCategory.js:76]: Please update more param for this function: policy, policyTags, policyCategories

**Comment by DylanDylann** [comment]
@dukenv0307 Can't access category page when creating request

https://github.com/Expensify/App/assets/141406735/aafa49f6-31cf-440d-9149-72ac077c506e

**Comment by DylanDylann** [comment]
`Bug`: Can't edit a category in request money. Error: Malformed currency code

https://github.com/Expensify/App/assets/141406735/036a2f73-9319-4062-bac3-80c45962063b

But I see this issue also happening on the main branch

File: src/pages/iou/request/step/IOURequestStepCategory.js [hunk 15]
```diff
@@ -101,6 +107,10 @@ function IOURequestStepCategory({
         navigateBack();
     };
 
+    if (!shouldShowCategories) {
+        return <FullPageNotFoundView shouldShow />;
```
Comment (DylanDylann) [inline src/pages/iou/request/step/IOURequestStepCategory.js:111]: What do you think about moving `FullPageNotFoundView` into `StepScreenWrapper`

**Comment by DylanDylann** [comment]
The changes look good. Testing

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by DylanDylann — APPROVED** [review]
LGTM

File: src/components/MoneyRequestConfirmationList.js [hunk 16]
```diff
@@ -767,10 +767,18 @@ function MoneyRequestConfirmationList(props) {
...
+                                            props.iouType,
+                                            props.transaction.transactionID,
+                                            props.reportID,
+                                            Navigation.getActiveRouteWithoutParams(),
+                                        ),
+                                    );
                                     return;
                                 }
-                                Navigation.navigate(ROUTES.MONEY_REQUEST_CATEGORY.getRoute(props.iouType, props.reportID));
+                                Navigation.navigate(ROUTES.MONEY_REQUEST_TAG.getRoute(props.iouType, props.reportID));
```
Comment (amyevans) [inline src/components/MoneyRequestConfirmationList.js:781]: What's up with this change?

File: src/libs/actions/IOU.ts [hunk 17]
```diff
@@ -299,16 +299,10 @@ function setMoneyRequestPendingFields(transactionID: string, pendingFields: Pend
...
 }
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-function setMoneyRequestCategory_temporaryForRefactor(transactionID: string, category: string) {
+function setMoneyRequestCategory(transactionID: string, category: string) {
     Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${transactionID}`, {category});
 }
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-function resetMoneyRequestCategory_temporaryForRefactor(transactionID: string) {
```
Comment (amyevans) [inline src/libs/actions/IOU.ts:308]: I'm still seeing an instance of this on your branch locally - perhaps from a merge with `main`?

File: src/libs/actions/IOU.ts [hunk 18]
```diff
@@ -299,16 +299,10 @@ function setMoneyRequestPendingFields(transactionID: string, pendingFields: Pend
     Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${transactionID}`, {pendingFields});
 }
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-function setMoneyRequestCategory_temporaryForRefactor(transactionID: string, category: string) {
```
Comment (amyevans) [inline src/libs/actions/IOU.ts:303]: Still seeing one usage of this too on your branch

File: src/pages/iou/request/step/StepScreenWrapper.js [hunk 19]
```diff
@@ -23,6 +24,9 @@ const propTypes = {
     /** Whether or not the wrapper should be shown (sometimes screens can be embedded inside another screen that already is using a wrapper) */
     shouldShowWrapper: PropTypes.bool.isRequired,
 
+    /** Whether or not display not found page */
```
Comment (amyevans) [inline src/pages/iou/request/step/StepScreenWrapper.js:27]:
```suggestion
    /** Whether or not to display not found page */
```

**Review by amyevans — CHANGES_REQUESTED** [review]
> Bug: Can't edit a category in request money. Error: Malformed currency code

I'm experiencing this as well

On `main`, I'm getting a different error:
> Uncaught TypeError: Cannot read properties of undefined (reading 'scrollTop')

so that's fun.

I don't feel comfortable approving/merging until the editing flow can be tested again, so let's wait until it's fixed in `main` ([Slack convo](https://expensify.slack.com/archives/C01GTK53T8Q/p1708384127377139)), then merge `main` and retest.

File: src/components/MoneyRequestConfirmationList.js (hunk 16 above)
Comment (DylanDylann) [reply src/components/MoneyRequestConfirmationList.js:781]: Thanks for your catch. @dukenv0307 It seems you update this in this [commit](https://github.com/Expensify/App/pull/35458/commits/6438c074738652dc38b7c4558f013ace0acab14c)

File: src/pages/iou/request/step/IOURequestStepCategory.js [hunk 20]
```diff
@@ -21,24 +32,50 @@ const propTypes = {
...
+    policyCategories,
 }) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
+    const isEditing = action === CONST.IOU.ACTION.EDIT;
+    const isEditingSplitBill = isEditing && iouType === CONST.IOU.TYPE.SPLIT;
+    const {category: transactionCategory} = ReportUtils.getTransactionDetails(isEditingSplitBill ? splitDraftTransaction : transaction);
+
+    const isPolicyExpenseChat = ReportUtils.isGroupPolicy(report);
+    const shouldShowCategories = isPolicyExpenseChat && (transactionCategory || OptionsListUtils.hasEnabledOptions(_.values(policyCategories)));
```
Comment (DylanDylann) [inline src/pages/iou/request/step/IOURequestStepCategory.js:78]: @dukenv0307 Should update to shouldShowNotFoundPage

**Comment by amyevans** [comment]
Okay great. Was able to test fully, looks good on that front! Let me just give the code another once over quick

**Review by amyevans — APPROVED** [review]

**Pull request merged**